scripts/Muon/GUI/ElementalAnalysis2/grouping_widget/ea_grouping_table_widget_view.py

```diff
@@ -0,0 +1,328 @@
+
+    def _context_menu_remove_group_action(self, slot):
+        if len(self._get_selected_row_indices()) > 1:
+            # use plural if >1 item selected
+            remove_group_action = QtWidgets.QAction('Remove Groups', self)
+        else:
+            remove_group_action = QtWidgets.QAction('Remove Group', self)
+        if self.num_rows() == 0:
+            remove_group_action.setEnabled(False)
+        remove_group_action.triggered.connect(slot)
+        return remove_group_action
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Slot connections
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def on_user_changes_group_name(self, slot):
+        self._validate_group_name_entry = slot
+
+    def on_remove_group_button_clicked(self, slot):
+        self.remove_group_button.clicked.connect(slot)
+
+    def on_table_data_changed(self, slot):
+        self._on_table_data_changed = slot
+
+    def on_cell_changed(self, _row, _col):
+        if not self._updating:
+            self._on_table_data_changed(_row, _col)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    #
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def get_table_item_text(self, row, col):
+        return self.grouping_table.item(row, col).text()
+
+    def get_table_item(self, row, col):
+        return self.grouping_table.item(row, col)
+
+    def set_to_analyse_state(self, row, state):
+        checked_state = QtCore.Qt.Checked if state is True else QtCore.Qt.Unchecked
+        item = self.get_table_item(row, inverse_group_table_columns['to_analyse'])
+        item.setCheckState(checked_state)
+
+    def set_to_analyse_state_quietly(self, row, state):
+        checked_state = QtCore.Qt.Checked if state is True else QtCore.Qt.Unchecked
+        item = self.get_table_item(row, inverse_group_table_columns['to_analyse'])
+        self.grouping_table.blockSignals(True)
+        item.setCheckState(checked_state)
+        self.grouping_table.blockSignals(False)
+
+    def get_table_contents(self):
+        if self._updating:
+            return []
+        ret = []
+        for row in range(self.num_rows()):
+            row_list = []
+            for col in range(self.num_cols()):
+                row_list.append(str(self.grouping_table.item(row, col).text()))
+            ret.append(row_list)
+        return ret
+
+    def clear(self):
+        # Go backwards to preserve indices
+        for row in reversed(range(self.num_rows())):
+            self.grouping_table.removeRow(row)
+
+    def on_rebin_combo_changed(self, index, row):
+        self.change_once = False
+        if index == 0:
```

Comment:
```suggestion
        self.grouping_table.setItem(row,4,QTableWidgetItem(str(index))
```
This is more concise than using 3 different if statements